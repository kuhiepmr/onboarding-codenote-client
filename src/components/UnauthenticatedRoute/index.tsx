import React from "react";
import { Redirect, Route } from "react-router-dom";
import { withAuthenticationStatus } from "../HOC";

const querystring = (name, url = window.location.href) => {
  name = name.replace(/[[]]/g, "\\$&");

  const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)", "i");
  const results = regex.exec(url);

  if (!results) {
    return null;
  }
  if (!results[2]) {
    return "";
  }

  return decodeURIComponent(results[2].replace(/\+/g, " "));
}

const UnauthenticatedRoute = ({ component: C, isAuthenticated, ...rest }) => {
  const redirect = querystring("redirect");
  return (
    <Route
      {...rest}
      render={props =>
        !isAuthenticated
          ? <C {...props} />
          : <Redirect
            to={redirect === "" || redirect === null ? "/" : redirect}
          />}
    />
  );
};

export default withAuthenticationStatus(UnauthenticatedRoute);
