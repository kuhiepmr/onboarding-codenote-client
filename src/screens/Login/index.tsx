import { Auth } from "aws-amplify";
import React, { Component } from "react";
import { ControlLabel, FormControl, FormGroup } from "react-bootstrap";
import LoaderButton from "../../components/LoaderButton";
import "./index.css";

class Login extends Component {
  state = {
    isLoading: false,
    email: "",
    password: "",
  };

  validateForm = () => (this.state.email.length && this.state.password.length)

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();

    this.setState({ isLoading: true });

    const { email: username, password } = this.state;

    try {
      await Auth.signIn({ username, password });

      this.handleLoginSuccess();
    } catch (e) {
      alert(e.message);
    }

    this.setState({ isLoading: false });
  };

  handleLoginSuccess = () => this.props.userHasAuthenticated(true);

  render() {
    return (
      <div className="Login">
        <form onSubmit={this.handleSubmit}>
          <FormGroup controlId="email" bsSize="large">
            <ControlLabel>Email</ControlLabel>
            <FormControl
              autoFocus
              type="email"
              value={this.state.email}
              onChange={this.handleChange}
            />
          </FormGroup>
          <FormGroup controlId="password" bsSize="large">
            <ControlLabel>Password</ControlLabel>
            <FormControl
              value={this.state.password}
              onChange={this.handleChange}
              type="password"
            />
          </FormGroup>
          <LoaderButton
            block
            bsSize="large"
            disabled={!this.validateForm()}
            type="submit"
            isLoading={this.state.isLoading}
            text="Login"
            loadingText="Logging in..."
          />
        </form>
      </div>
    );
  }
}

export default Login;
