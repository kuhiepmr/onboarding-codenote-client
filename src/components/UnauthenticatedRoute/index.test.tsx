import { mount, shallow } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import Login from '../../screens/Login/index';
import UnauthenticatedRoute from './index';

test('renders without crashing', () => {
  const initState = {
    authenticate: { isAuthenticated: true }
  }
  const store = configureStore([])(initState)
  const login = mount(<Login store={store} />);
  const wrapper = shallow(<UnauthenticatedRoute store={store} component={login} />);

  expect(wrapper).toMatchSnapshot();
});