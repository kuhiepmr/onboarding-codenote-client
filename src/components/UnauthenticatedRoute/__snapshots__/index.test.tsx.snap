// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<UnauthenticatedRoute
  component={
    <Connect(Login)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Login
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        userHasAuthenticated={[Function]}
      >
        <div
          className="Login"
        >
          <form
            onSubmit={[Function]}
          >
            <FormGroup
              bsClass="form-group"
              bsSize="large"
              controlId="email"
            >
              <div
                className="form-group form-group-lg"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                    htmlFor="email"
                  >
                    Email
                  </label>
                </ControlLabel>
                <FormControl
                  autoFocus={true}
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  type="email"
                  value=""
                >
                  <input
                    autoFocus={true}
                    className="form-control"
                    id="email"
                    onChange={[Function]}
                    type="email"
                    value=""
                  />
                </FormControl>
              </div>
            </FormGroup>
            <FormGroup
              bsClass="form-group"
              bsSize="large"
              controlId="password"
            >
              <div
                className="form-group form-group-lg"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                    htmlFor="password"
                  >
                    Password
                  </label>
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  type="password"
                  value=""
                >
                  <input
                    className="form-control"
                    id="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </FormControl>
              </div>
            </FormGroup>
            <LoaderButton
              block={true}
              bsSize="large"
              disabled={true}
              isLoading={false}
              loadingText="Logging in ..."
              text="Login"
              type="submit"
            >
              <Button
                active={false}
                block={true}
                bsClass="btn"
                bsSize="large"
                bsStyle="default"
                className="LoaderButton "
                disabled={true}
                type="submit"
              >
                <button
                  className="LoaderButton  btn btn-lg btn-default btn-block"
                  disabled={true}
                  type="submit"
                >
                  Login
                </button>
              </Button>
            </LoaderButton>
          </form>
        </div>
      </Login>
    </Connect(Login)>
  }
  dispatch={[Function]}
  isAuthenticated={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
